### DISCUSSION TITLE
Ideas and Inspirations to look upto

### DISCUSSION THREAD

**Discussion by omar391** [livebud/bud#249] [Ideas]
Hi Bud team,

I have some "feature"s suggestions from the following (not in any particular order, apologize if it's already suggested):

# Marko-js V5+
 - Server-side streaming
 - Progressive rendering 
 - Partial Hydration (added doc links for possible impl ways )
 - https://markojs.com/

# Inertia Js
 - Act as a glue layer between "Svelte" and other Backend languages i.e. Laravel, Rails, etc
 - Also has client and server-side routing.
 - Most importantly, only send the minimal hydration data in a JSON format in successive calls after the initial render.  
 - https://inertiajs.com/

# Others
- Compile Go to "wasm" and share common code with backend and frontend.
- Use the "Go" file as a JS module: https://github.com/teamortix/golang-wasm
- Resumeable hydration: https://qwik.builder.io/

 ### Blog posts: 
- https://dev.to/ryansolid/marko-compiling-fine-grained-reactivity-4lk4
- https://dev.to/this-is-learning/patterns-for-building-javascript-websites-in-2022-5a93
- https://www.npmjs.com/package/@elo7/fastify-svelte-partial-hydration
- https://qwik.builder.io/docs/concepts/resumable

**Comment by matthewmueller** [comment]
Hey @omar391, thanks for the research! I'm planning on abstracting out the view renderer a bit further to allow for at least some alternatives like this. I'm definitely interested in figuring out ways to ship less JSON down for the hydration step!

**Reply by omar391** [reply]
@matthewmueller Looking forward to an RFC or some form of a milestone! <3.

Also this would be a good starting point: https://qwik.builder.io/docs/concepts/resumable
